src/dfx/src/commands/deploy.rs

```diff
@@ -156,6 +175,14 @@ pub fn exec(env: &dyn Environment, opts: DeployOpts) -> DfxResult {
 
     let call_sender = CallSender::from(&opts.wallet)
         .map_err(|e| anyhow!("Failed to determine call sender: {}", e))?;
+    let created_at_time = opts.created_at_time.or_else(|| {
+        let now = SystemTime::now()
+            .duration_since(UNIX_EPOCH)
+            .unwrap()
+            .as_nanos() as u64;
+        info!(env.get_logger(), "created-at-time is {now}.");
```

Comment:
in the case where the user retries by running the command again (with the `--created-at-time option), they're going to need to know which canister to retry for, in case the canister settings differ.  (We might be able to get away with retrying only the first canister to be created, or we might not.)